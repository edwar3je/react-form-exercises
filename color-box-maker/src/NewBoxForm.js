import { useState } from 'react';
import './NewBoxForm.css';

const NewBoxForm = ({ addBox }) => {
    
    const initialState = {
        height: "",
        width: "",
        backgroundColor: ""
    };

    const colorInput = document.querySelector('input.backgroundColor');

    const resetColor = () => {
        colorInput.value = "";
    };
    
    const [formData, setFormData] = useState(initialState);
    const handleChange = e => {
        const { name, value } = e.target;
        setFormData(formData => ({
            ...formData,
            [name]: value
        }))
    }; 

    const handleSubmit = (e) => {
        e.preventDefault();
        const { height, width, backgroundColor} = formData;
        addBox(height, width, backgroundColor);
        alert(`Created ${backgroundColor} box that's ${height} high and ${width} wide`);
        setFormData(initialState);
        resetColor();
    };
    
    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor="width">Width</label>
            <input type="text" id="width" name="width" placeholder="width" value={formData.width} onChange={handleChange}></input>
            <p>{formData.width}</p>
            <label htmlFor="height">Height</label>
            <input type="text" id="height" name="height" placeholder="height" value={formData.height} onChange={handleChange}></input>
            <p>{formData.height}</p>
            <label htmlFor="backgroundColor">Background Color</label>
            <input type="text" className="backgroundColor" id="backgroundColor" name="backgroundColor" placeholder="background color" value={formData.value} onChange={handleChange}></input>
            <p>{formData.backgroundColor}</p>
            <button type="submit">Add New Box</button>
        </form>
    )
}

export default NewBoxForm;