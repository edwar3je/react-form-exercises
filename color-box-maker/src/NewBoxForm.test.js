import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import NewBoxForm from './NewBoxForm';

it("renders without crashing", () => {
    render(
        <NewBoxForm 
            addBox={() => console.log("you added a box")}
        />
    );
});

it("matches the snapshot", () => {
    const { asFragment } = render(
        <NewBoxForm 
            addBox={() => console.log("you added a box")}
        />
    );
    expect(asFragment()).toMatchSnapshot();
});