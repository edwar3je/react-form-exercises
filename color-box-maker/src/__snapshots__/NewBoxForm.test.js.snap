// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="width"
    >
      Width
    </label>
    <input
      id="width"
      name="width"
      placeholder="width"
      type="text"
      value=""
    />
    <p />
    <label
      for="height"
    >
      Height
    </label>
    <input
      id="height"
      name="height"
      placeholder="height"
      type="text"
      value=""
    />
    <p />
    <label
      for="backgroundColor"
    >
      Background Color
    </label>
    <input
      class="backgroundColor"
      id="backgroundColor"
      name="backgroundColor"
      placeholder="background color"
      type="text"
      value=""
    />
    <p />
    <button
      type="submit"
    >
      Add New Box
    </button>
  </form>
</DocumentFragment>
`;
